### ISSUE TITLE
connection.unsafeDestroy() and connection.close() hang forever

### ISSUE THREAD

**Issue by webhype** [cody-greene/node-rabbitmq-client#40]
On Mac OS, my code works fine, including `connection.close()`; however, on Linux the two connection closing calls `connection.unsafeDestroy()` and `connection.close()` hang forever. Everything else in terms of dealing with the RabbitMQ queue seems to work just fine before that point.

The Linux code is embedded in a Vercel `pkg` executable, if that matters. Sometimes strange things can happen when packaging NodeJS code in `pkg`, `electron` etc.

I am not saying there's a bug in your library, just looking for pointers to get to the bottom of possible scenarios / deeper reasons that closing the socket to `127.0.0.1`, port `5672` could hang indefinitely. Thanks!

**Comment by cody-greene** [comment]
This is not much to go on. Can you provide a simple standalone script which demonstrates this issue (on your machines)?

Note that `.close()` normally waits for you to close any channels, publishers, and consumers first.

-------- Original Message --------

**Comment by cody-greene** [comment]
Ideally you could just attach a debugger and step through each line of the `close()` method. I have no idea if you can do that with the executable you're producing, so worst case, just patch the method with some console logs. Copy-paste the source method and monkey-patch it before running your build or edit the file in node_modules, whatever.

Either way you'll need to narrow down where the method is getting stuck.

**Comment by webhype** [comment]
Thanks, I'll report back to you; I suspect I opened some resources and didn't close them. My program exits "apparently fine" when I `Promise.race()` the `connection.close()` with a `setTimeoutPromise(5000)` so... it's probably exactly as you said, I didn't close something that I opened... when I have found the culprit I will post here and close the issue.

**Comment by Bugs5382** [comment]
So... I have only noticed this error when running unit testing. (I am the author of [fastify-rabbitmq](https://github.com/Bugs5382/fastify-rabbitmq) which uses this package as it's app.). 

I have never seen a unit test inside this package that tests this scenario, where the domain/ip of the RabbitMQ server does not exist, so I had to make one up for mine and just to check if it breaks out of the try/catch code. This only happens, as far as I can tell, on initial connection thought. Not after an initial connection is made and a reconnection phase is going on.
